Mutual regulation of tumour vessel normalization and immunostimulatory reprogramming
======

This repository contains the source data and source codes for bioinformatic analysis. The manuscript has been accepted by [Nature](http://www.nature.com/nature/index.html).

File Name  | Description
------------- | -------------
**data** | source data for the bioinformatic analysis
**LungMetQuantify** | [MATLAB](https://www.mathworks.com/) source code for lung macro-metastasis quantification
**VesselQuantify** | [CellProfiler](http://cellprofiler.org/) pipelines for vessel density/length and co-localization quantifications
**SUM_PCA** | compare ∑GPAGs − ∑PPAGs with the first principle component from Principle Component Analysis (**Supplementary Table 1c-f**)
**REVIGO_Plot** | source codes for summarizing GPAGs and PPAGs using [REVIGO](http://revigo.irb.hr/) plots (**Supplementary Table 2**)
**IF_Visualization** | source codes of dot plots for immuno-staining data visualization
**Immune_Profiling** | flow panel design for the immune profiling
**Fig1** | source codes for **Fig. 1a,b,f**, and **Supplementary Table 1a,b** and **Supplementary Table 3**
**Fig4** | source codes for **Fig. 4a,b,i,j**
**ExtendedFig1** | source codes for **Extended Data Fig. 1b-e**
**ExtendedFig3** | source codes for **Extended Data Fig. 3q**
**ExtendedFig7** | source codes for **Extended Data Fig. 7b-h**
**ExtendedFig8** | source codes for **Extended Data Fig. 8c,d**

## Source Code/Data Download

You can use `git` command tool to clone the whole repository.

```
git clone https://github.com/lintian0616/vesselNormalization.git
```

Alternatively, you can click **Download ZIP** and later unzip the file in your local computer.

Should you have any question, please do not hesitate to contact Lin Tian (lintian0616@gmail.com) and cc to Dr. Xiang Zhang (xiangz@bcm.edu).

## Acknowledgement

We thank the following bioinformatic resources for their support on our research.

* [CyVerse](http://www.cyverse.org/) provides us with computational systems for bioinformatic analysis. All the R programming codes were run on the instance called **Data Mining for Cancer Biology**.
* [GenePattern](https://genepattern.broadinstitute.org/) allows us perform all the Gene Set Enrichment Analysis (**GSEA**) and single sample GSEA projection (**ssGSEA**).
* [GenomeSpace](http://www.genomespace.org/) helps us organize our biological data.
* [GitHub](https://github.com/) and [Bitbucket](https://bitbucket.org) helps us organize our source codes.
